src/ixdat/plotters/ms_plotter.py

```diff
@@ -397,6 +482,438 @@ def _parse_overloaded_inputs(
+            sort_spectra (list or str): Whether or not to sort the spectra data prior to
+                plotting.
+                There is three specifers:
+                    'none':
+                        This gives no new sorting. Effectively the spectras are sorted
+                        by time. (tstamp for each of the spectrum).
+                    'linear' (default):
+                        the spectras are sorted linear to v_name from low to high.
+                    a list of same shape as field.data to be sorted:
+                        This list is passed directly as the indices to sort the spectras.
+                        Defaults to sort lowest to highest value. Example
+                Note: If tspan spans a time span of a measurement with up and down
+                cycles in v_name, this might yield funny looking heat_plots.
+
+                    Example:
+                    Scanning up and down in temperature from T_low to T_high the spectras
+                    obtained wil be plotted from [T_low_start ..  T_high .. T_low_end].
+
+                    - If 'none' sorting is specified leads to heat_plot xaxis linearly
+                    from T_low_start to T_low_end missing representation of the high
+                    values in the middle of the axis.
+
+                    - If 'linear' sorting is specified all spectras obtained
+                        are sorted linearly from lowest v_name_value to highest v_name.
+                        When data is assymetric from scanning up or down in v_name this
+                        leads to abrupt looking figures since two non similair spectras
+                        are obtained at similair v_name value and hence plotted next to
+                        eachother.
+
+            kwargs: extra key-word args are passed on to matplotlib's plot()
+        """
+
+        if logplot is None:
+            logplot = not mol_lists and not mass_lists
+
+        if not axes:
+            if ms_data == "top":
+                n_bottom = 1
+                n_top = 2 if (mass_lists or mol_lists) else 1
+                ms_axes = 0
+                ms_spec_axes = 1
+            else:
+                n_top = 1
+                n_bottom = 2 if (mass_lists or mol_lists) else 1
+                ms_axes = 1
+                ms_spec_axes = 0
+
+            axes = self.new_two_panel_axes(
+                n_bottom=n_bottom,
+                n_top=n_top,
+                emphasis=emphasis,
+            )
+
+        measurement = measurement or self.measurement
+
+        if (
+            mass_list
+            or mass_lists
+            or mol_list
+            or mol_lists
+            or hasattr(measurement, "mass_list")
+        ):
+            # then we have MS data!
+            self.ms_plotter.plot_vs(
+                x_name=vs_name,
+                measurement=measurement,
+                axes=[axes[ms_axes], axes[2]]
+                if (mass_lists or mol_lists)
```

Comment:
Can you define axes in an earlier line to avoid this syntax mess?

Reply:
I have defined the ms_plot_axes earlier know.